core/src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -416,6 +421,21 @@ void validateIncomingState(ClusterState incomingState, ClusterState lastSeenClus
             logger.warn(message);
             throw new IllegalStateException(message);
         }
+
+        final ClusterState state = clusterStateSupplier.get();
```

Comment:
I wonder if we should strengthen ZenDiscovery.shouldIgnoreOrRejectNewClusterState to check for version equality (and assert uuid) and use it [in line 410](https://github.com/elastic/elasticsearch/pull/17038/commits/cffc315dca66fa6af109a7cb5768acfb05e140f7#diff-594996d316ac34afe009d496db4abe34R410)  then all of this can be change to just throwing an exception if it doesn't like it.